### DISCUSSION TITLE
Mark

### DISCUSSION THREAD

**Discussion by wartburggraaf** [nl-design-system/backlog#332] [Component suggestions]
<img width="480" alt="visuele weergave van een Mark component" src="https://github.com/user-attachments/assets/1d6ca4c2-2452-4ef1-93b3-df87431d66d4">

We willen graag meer leren over de Mark component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook, Github of screenshots. Met deze informatie kunnen we bewijzen dat de component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1725369675413979)

---

## Naam

Mark

## Doel

Markeert tekst, zodat je ernaar kan verwijzen vanuit een andere context of omdat de tekst relevant is voor de huidige activiteit.

## Ook bekend als

Highlight

## Gerelateerde componenten

[Strong](https://github.com/orgs/nl-design-system/discussions/291)

## Richtlijnen
Acceptatiecriteria: [NL Design System Acceptatiecriteria voor Mark](https://nldesignsystem.nl/mark)

## Onderzoek

[`<mark>`: The Mark Text element - MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/mark)
Documentatie van Mozilla Developers Network over `<mark>` HTML element.

[Tweaking Text Level Styles - Adrian Roselli](https://adrianroselli.com/2017/12/tweaking-text-level-styles.html)
Adrian Roselli beschrijft waar je zoal rekening mee moet houden bij de Mark component.

[Short note on making your mark (more accessible)](https://www.tpgi.com/short-note-on-making-your-mark-more-accessible/)
Steve Faulkner beschreef wat je kun doen om de HTML5 `mark` component toegankelijker te maken. Maar inmiddels lijkt deze [betere support](https://html5accessibility.com/#text:~:text=collapse-,mark,-Accessibly%20supported) te hebben waardoor dit wellicht niet meer nodig is.

**Comment by jeffreylauwers** [comment]
## Amsterdam

- [Storybook](https://designsystem.amsterdam/?path=/docs/components-text-mark--docs)
- [Figma](https://www.figma.com/design/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?node-id=2927-29175)
- [GitHub](https://github.com/Amsterdam/design-system/tree/develop/packages/css/src/components/mark)

---

### Screenshots / Varianten

**Storybook**
![Scherm­afbeelding 2024-09-05 om 13 00 17](https://github.com/user-attachments/assets/56084bf6-1c77-4dfd-b138-a77592b86e76)

---

### Notities

- Gele achtergrondkleur.
- Documentatie beschikbaar (Storybook).

**Comment by jeffreylauwers** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-mark--docs)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/mark)

---

### Screenshots / Varianten

**Storybook**

Default
Markup with the `<mark>` element. Styling via `utrecht-mark` class name.
![Scherm­afbeelding 2024-09-05 om 13 35 58](https://github.com/user-attachments/assets/dcd387e2-9b45-4166-a5f8-c5a086ceb4c5)

Alternate markup
With the `<span>` element the styling should work too. Note: using the `<mark>` element is preferred, because it works better when CSS can not be loaded.
![Scherm­afbeelding 2024-09-05 om 13 37 21](https://github.com/user-attachments/assets/50621dd8-cce6-4af4-9f29-aa935de20471)

Appearance in print
Print styling for the mark component. This styling is automatic via a media query, but for testing it can be simulated using the `utrecht-mark--print` class name.
![Scherm­afbeelding 2024-09-05 om 13 37 43](https://github.com/user-attachments/assets/7e01ff2e-2515-432b-a889-4b8a1ca9eab6)


Windows High Contrast mode
High contrast styling for the mark component. This styling is automatic via a media query, but for testing it can be simulated using the `utrecht-mark--windows-high-contrast` class name.
![Scherm­afbeelding 2024-09-05 om 13 38 22](https://github.com/user-attachments/assets/df4b263f-0081-41e0-82e7-a1ae18c504ca)

---

### Notities

- Gele achtergrondkleur.
- Verschillende varianten.
- Documentatie beschikbaar (Storybook).

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Mark op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=mark)

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Community ✨

Help je mee hem door de Candidate stappen te krijgen?
[Mark op het Candidate bord](https://github.com/orgs/nl-design-system/projects/32/views/1?filterQuery=mark)

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Candidate ✨

Help je mee hem door de Hall of Fame stappen te krijgen?
[Mark op het Hall of Fame bord](https://github.com/orgs/nl-design-system/projects/30/views/1?filterQuery=mark)

**Comment by Robbert** [comment]
Ik ben de Mark tegengekomen binnen een Combobox component, bij de website van [gemeente Borger-Odoorn](https://www.borger-odoorn.nl/actueel-overzicht):

<img width="601" alt="In de resultaten van een combobox is het woord 'gemeente' gemarkeerd" src="https://github.com/user-attachments/assets/2b6ef2ac-1d38-49ad-96af-8062739c9fcb" />
